lib/rbs_rails/active_record.rb

```diff
@@ -336,65 +329,12 @@ def authenticate_#{attribute}: (String) -> (#{klass_name} | false)
       private def enum_scope_methods(singleton:)
         # @type var methods: Array[String]
         methods = []
-        enum_definitions.each do |hash|
-          hash.each do |name, values|
-            next if IGNORED_ENUM_KEYS.include?(name)
-
-            values.each do |label, value|
-              value_method_name = enum_method_name(hash, name, label)
-              methods << "def #{singleton ? 'self.' : ''}#{value_method_name}: () -> #{relation_class_name}"
-            end
-          end
+        klass.enum_definitions.each do |_, method_name|
+          methods << "def #{singleton ? 'self.' : ''}#{method_name}: () -> #{relation_class_name}"
         end
         methods.join("\n")
       end
 
-      private def enum_definitions
-        @enum_definitions ||= build_enum_definitions
-      end
-
-      # We need static analysis to detect enum.
-      # ActiveRecord has `defined_enums` method,
-      # but it does not contain _prefix and _suffix information.
-      private def build_enum_definitions
-        ast = parse_model_file
-        return [] unless ast
-
-        traverse(ast).map do |node|
-          # @type block: nil | Hash[untyped, untyped]
-          next unless node.type == :send
-          next unless node.children[0].nil?
-          next unless node.children[1] == :enum
-
-          definitions = node.children[2]
-          next unless definitions
-          next unless definitions.type == :hash
```

Comment:
I found the current implementation only supports hash-style enums.
But Rails also allows defining an enum via array-style.
https://api.rubyonrails.org/v6.1.7.10/classes/ActiveRecord/Enum.html

The new implementation also does not support array-style.
I'll work on it the later step. Let's go as is in this PR because there is no regression.